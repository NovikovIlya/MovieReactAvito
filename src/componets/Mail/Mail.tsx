import styles from './Mail.module.scss';
import { SearchOutlined } from '@ant-design/icons';
import React, { useEffect, useRef, useState } from 'react';
import Highlighter from 'react-highlight-words';
import type { InputRef } from 'antd';
import { Button, Input, Space, Table } from 'antd';
import type { ColumnType, ColumnsType } from 'antd/es/table';
import type { FilterConfirmProps } from 'antd/es/table/interface';
import {
  useDeleteMessageMutation,
  useGetMessageMutation,
} from '../../store/MovieApi';
import { Link } from 'react-router-dom';
import { useAppSelector } from '../../hooks/redux';
import { DataType } from '../../types';



type DataIndex = keyof DataType;

const Mail = () => {
  const myName = useAppSelector((state) => state.sliceMovie.myName);
  const [searchText, setSearchText] = useState('');
  const [searchedColumn, setSearchedColumn] = useState('');
  const searchInput = useRef<InputRef>(null);
  const [getMessage, { data: dataMessage }] = useGetMessageMutation();
  const [deleteMessage] = useDeleteMessageMutation();
  const [users, setUsers] = useState([]);
  const data: any = users;
  const getColumnSearchProps = (dataIndex: DataIndex): ColumnType<DataType> => ({
    filterDropdown: ({ setSelectedKeys, selectedKeys, confirm, clearFilters, close }) => (
      <div style={{ padding: 8 }} onKeyDown={(e) => e.stopPropagation()}>
        <Input
          ref={searchInput}
          placeholder={`Search ${dataIndex}`}
          value={selectedKeys[0]}
          onChange={(e) => setSelectedKeys(e.target.value ? [e.target.value] : [])}
          onPressEnter={() => handleSearch(selectedKeys as string[], confirm, dataIndex)}
          style={{ marginBottom: 8, display: 'block' }}
        />
        <Space>
          <Button
            type="primary"
            onClick={() => handleSearch(selectedKeys as string[], confirm, dataIndex)}
            icon={<SearchOutlined />}
            size="small"
            style={{ width: 90 }}>
            Search
          </Button>
          <Button
            onClick={() => clearFilters && handleReset(clearFilters)}
            size="small"
            style={{ width: 90 }}>
            Reset
          </Button>
          <Button
            type="link"
            size="small"
            onClick={() => {
              confirm({ closeDropdown: false });
              setSearchText((selectedKeys as string[])[0]);
              setSearchedColumn(dataIndex);
            }}>
            Filter
          </Button>
          <Button
            type="link"
            size="small"
            onClick={() => {
              close();
            }}>
            close
          </Button>
        </Space>
      </div>
    ),
    filterIcon: (filtered: boolean) => (
      <SearchOutlined style={{ color: filtered ? '#1677ff' : undefined }} />
    ),
    onFilter: (value, record) =>
      record[dataIndex]
        .toString()
        .toLowerCase()
        .includes((value as string).toLowerCase()),
    onFilterDropdownOpenChange: (visible) => {
      if (visible) {
        setTimeout(() => searchInput.current?.select(), 100);
      }
    },
    render: (text) =>
      searchedColumn === dataIndex ? (
        <Highlighter
          highlightStyle={{ backgroundColor: '#ffc069', padding: 0 }}
          searchWords={[searchText]}
          autoEscape
          textToHighlight={text ? text.toString() : ''}
        />
      ) : (
        text
      ),
  });
  const columns: ColumnsType<any> = [
    {
      title: 'Username',
      dataIndex: 'username',
      key: 'username',
      width: '20%',
      
      // render: (text) => <a href='/'>{text}</a>,
      //@ts-ignore
      ...getColumnSearchProps('username'),
    },
    {
      title: 'Theme',
      dataIndex: 'theme',
      key: 'theme',
      width: '20%',
      //@ts-ignore
      ...getColumnSearchProps('theme'),
    },
    {
      title: 'Date',
      dataIndex: 'date',
      key: 'date',
      width: '10%',
      //@ts-ignore
      ...getColumnSearchProps('date'),
    },
    {
      title: 'Time',
      dataIndex: 'time',
      key: 'time',
      width: '10%',
      //@ts-ignore
      ...getColumnSearchProps('time'),
    },
    {
      title: 'Delete',
      dataIndex: 'delete',
      key: 'delete',
      width: '10%',
      //@ts-ignore
      // ...getColumnSearchProps('delete'),
    },
    {
      title: 'Read',
      dataIndex: 'read',
      key: 'read',
      width: '10%',
      //@ts-ignore
      ...getColumnSearchProps('read'),
    },
    {
      title: 'Link',
      dataIndex: 'link',
      key: 'link',
      width: '10%',
    },
  ];

  //hooks
  useEffect(() => {
    if (dataMessage) {
      const dataArray = dataMessage.map((item) => {
        return {
          key: item.id,
          username: item.myname,
          theme: item.theme,
          date: item.date,
          time: item.time,
          delete: (
            <Button onClick={deleteMessageFn} name={item.id}>
              Delete
            </Button>
          ),
          read: item.read ? 'yes' : 'no',
          link: <Link to={`/onemail/${item.id}/${item.myname}`}>Go</Link>,
        };
      });
      const uniqueArray = [];
      dataArray.forEach((element) => {
        if (!uniqueArray.some((e) => e.theme === element.theme)) {
          uniqueArray.push(element);
        }
      });
      setUsers(uniqueArray.reverse());
    }
  }, [dataMessage]);

  useEffect(() => {
    getMessage({ username: myName });
  }, []);

  //functions
  const handleSearch = (
    selectedKeys: string[],
    confirm: (param?: FilterConfirmProps) => void,
    dataIndex: DataIndex,
  ) => {
    confirm();
    setSearchText(selectedKeys[0]);
    setSearchedColumn(dataIndex);
  };
  const handleReset = (clearFilters: () => void) => {
    clearFilters();
    setSearchText('');
  };
  const deleteMessageFn = (event) => {
    deleteMessage({
      username: myName,
      id: event.target.name,
    });
    getMessage({ username: myName });
  };

  return (
    <div>
      <div className={styles.tabl}>
        <Table columns={columns} dataSource={data} />
      </div>
    </div>
  );
};

export default Mail;
