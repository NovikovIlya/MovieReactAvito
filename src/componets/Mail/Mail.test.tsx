import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import Mail from './Mail';
import { Provider } from 'react-redux';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

describe('test component mail', () => {
  it('renders mail', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const mail = render(
      <Provider store={store}>
        <BrowserRouter>
          <Mail />
        </BrowserRouter>
      </Provider>,
    );
    // eslint-disable-next-line testing-library/no-debugging-utils
    screen.debug();
    expect(mail).toMatchSnapshot()
    expect(screen.getByText('Username')).toBeInTheDocument()
  });
});
