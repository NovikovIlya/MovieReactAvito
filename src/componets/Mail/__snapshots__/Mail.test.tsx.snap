// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component mail renders mail 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="tabl"
        >
          <div
            class="ant-table-wrapper css-dev-only-do-not-override-i0102m"
          >
            <div
              class="ant-spin-nested-loading css-dev-only-do-not-override-i0102m"
            >
              <div
                class="ant-spin-container"
              >
                <div
                  class="ant-table ant-table-empty"
                >
                  <div
                    class="ant-table-container"
                  >
                    <div
                      class="ant-table-content"
                    >
                      <table
                        style="table-layout: auto;"
                      >
                        <colgroup>
                          <col
                            style="width: 30%;"
                          />
                          <col
                            style="width: 20%;"
                          />
                          <col
                            style="width: 20%;"
                          />
                          <col
                            style="width: 20%;"
                          />
                          <col
                            style="width: 20%;"
                          />
                          <col
                            style="width: 20%;"
                          />
                          <col
                            style="width: 10%;"
                          />
                        </colgroup>
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              <div
                                class="ant-table-filter-column"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Username
                                </span>
                                <span
                                  class="ant-dropdown-trigger ant-table-filter-trigger"
                                  role="button"
                                  tabindex="-1"
                                >
                                  <span
                                    aria-label="search"
                                    class="anticon anticon-search"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="search"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              <div
                                class="ant-table-filter-column"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Theme
                                </span>
                                <span
                                  class="ant-dropdown-trigger ant-table-filter-trigger"
                                  role="button"
                                  tabindex="-1"
                                >
                                  <span
                                    aria-label="search"
                                    class="anticon anticon-search"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="search"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              <div
                                class="ant-table-filter-column"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Date
                                </span>
                                <span
                                  class="ant-dropdown-trigger ant-table-filter-trigger"
                                  role="button"
                                  tabindex="-1"
                                >
                                  <span
                                    aria-label="search"
                                    class="anticon anticon-search"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="search"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              <div
                                class="ant-table-filter-column"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Time
                                </span>
                                <span
                                  class="ant-dropdown-trigger ant-table-filter-trigger"
                                  role="button"
                                  tabindex="-1"
                                >
                                  <span
                                    aria-label="search"
                                    class="anticon anticon-search"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="search"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              <div
                                class="ant-table-filter-column"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Delete
                                </span>
                                <span
                                  class="ant-dropdown-trigger ant-table-filter-trigger"
                                  role="button"
                                  tabindex="-1"
                                >
                                  <span
                                    aria-label="search"
                                    class="anticon anticon-search"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="search"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              <div
                                class="ant-table-filter-column"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Read
                                </span>
                                <span
                                  class="ant-dropdown-trigger ant-table-filter-trigger"
                                  role="button"
                                  tabindex="-1"
                                >
                                  <span
                                    aria-label="search"
                                    class="anticon anticon-search"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="search"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              Link
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        >
                          <tr
                            class="ant-table-placeholder"
                          >
                            <td
                              class="ant-table-cell"
                              colspan="7"
                            >
                              <div
                                class="css-dev-only-do-not-override-i0102m ant-empty ant-empty-normal"
                              >
                                <div
                                  class="ant-empty-image"
                                >
                                  <svg
                                    height="41"
                                    viewBox="0 0 64 41"
                                    width="64"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <g
                                      fill="none"
                                      fill-rule="evenodd"
                                      transform="translate(0 1)"
                                    >
                                      <ellipse
                                        cx="32"
                                        cy="33"
                                        fill="#f5f5f5"
                                        rx="32"
                                        ry="7"
                                      />
                                      <g
                                        fill-rule="nonzero"
                                        stroke="#d9d9d9"
                                      >
                                        <path
                                          d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                        />
                                        <path
                                          d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                          fill="#fafafa"
                                        />
                                      </g>
                                    </g>
                                  </svg>
                                </div>
                                <div
                                  class="ant-empty-description"
                                >
                                  No data
                                </div>
                              </div>
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="tabl"
      >
        <div
          class="ant-table-wrapper css-dev-only-do-not-override-i0102m"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-i0102m"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table ant-table-empty"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup>
                        <col
                          style="width: 30%;"
                        />
                        <col
                          style="width: 20%;"
                        />
                        <col
                          style="width: 20%;"
                        />
                        <col
                          style="width: 20%;"
                        />
                        <col
                          style="width: 20%;"
                        />
                        <col
                          style="width: 20%;"
                        />
                        <col
                          style="width: 10%;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            <div
                              class="ant-table-filter-column"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Username
                              </span>
                              <span
                                class="ant-dropdown-trigger ant-table-filter-trigger"
                                role="button"
                                tabindex="-1"
                              >
                                <span
                                  aria-label="search"
                                  class="anticon anticon-search"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            <div
                              class="ant-table-filter-column"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Theme
                              </span>
                              <span
                                class="ant-dropdown-trigger ant-table-filter-trigger"
                                role="button"
                                tabindex="-1"
                              >
                                <span
                                  aria-label="search"
                                  class="anticon anticon-search"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            <div
                              class="ant-table-filter-column"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Date
                              </span>
                              <span
                                class="ant-dropdown-trigger ant-table-filter-trigger"
                                role="button"
                                tabindex="-1"
                              >
                                <span
                                  aria-label="search"
                                  class="anticon anticon-search"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            <div
                              class="ant-table-filter-column"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Time
                              </span>
                              <span
                                class="ant-dropdown-trigger ant-table-filter-trigger"
                                role="button"
                                tabindex="-1"
                              >
                                <span
                                  aria-label="search"
                                  class="anticon anticon-search"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            <div
                              class="ant-table-filter-column"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Delete
                              </span>
                              <span
                                class="ant-dropdown-trigger ant-table-filter-trigger"
                                role="button"
                                tabindex="-1"
                              >
                                <span
                                  aria-label="search"
                                  class="anticon anticon-search"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            <div
                              class="ant-table-filter-column"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Read
                              </span>
                              <span
                                class="ant-dropdown-trigger ant-table-filter-trigger"
                                role="button"
                                tabindex="-1"
                              >
                                <span
                                  aria-label="search"
                                  class="anticon anticon-search"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Link
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-placeholder"
                        >
                          <td
                            class="ant-table-cell"
                            colspan="7"
                          >
                            <div
                              class="css-dev-only-do-not-override-i0102m ant-empty ant-empty-normal"
                            >
                              <div
                                class="ant-empty-image"
                              >
                                <svg
                                  height="41"
                                  viewBox="0 0 64 41"
                                  width="64"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <g
                                    fill="none"
                                    fill-rule="evenodd"
                                    transform="translate(0 1)"
                                  >
                                    <ellipse
                                      cx="32"
                                      cy="33"
                                      fill="#f5f5f5"
                                      rx="32"
                                      ry="7"
                                    />
                                    <g
                                      fill-rule="nonzero"
                                      stroke="#d9d9d9"
                                    >
                                      <path
                                        d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                      />
                                      <path
                                        d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                        fill="#fafafa"
                                      />
                                    </g>
                                  </g>
                                </svg>
                              </div>
                              <div
                                class="ant-empty-description"
                              >
                                No data
                              </div>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
